modules/javafx.web/src/main/java/com/sun/webkit/network/PublicSuffixes.java

```diff
@@ -78,94 +84,116 @@ static boolean isPublicSuffix(String domain) {
         if (domain.length() == 0) {
             return false;
         }
-        Rule rule = RULES.get(domain);
-        if (rule == Rule.EXCEPTION_RULE) {
-            return false;
-        } else if (rule == Rule.SIMPLE_RULE || rule == Rule.WILDCARD_RULE) {
-            return true;
-        } else {
-            int pos = domain.indexOf('.') + 1;
-            if (pos == 0) {
-                pos = domain.length();
-            }
-            String parent = domain.substring(pos);
-            return RULES.get(parent) == Rule.WILDCARD_RULE;
-        }
+
+        Rules rules = Rules.getRules(domain);
+        return rules == null ? false : rules.match(domain);
```

Comment:
This will return `false` if there is no `public_suffix_list.dat` file or if there is an error parsing it. Can you confirm that this is the desired fallback behavior?